File: app/src/main/java/com/lence/penguinsvskillerwhales/view/MainActivity.java
package com.lence.penguinsvskillerwhales.view;

import android.app.ProgressDialog;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Button;
import android.widget.FrameLayout;

import com.lence.penguinsvskillerwhales.R;
import com.lence.penguinsvskillerwhales.life.Presenter;
import com.lence.penguinsvskillerwhales.model.Organism;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity implements UpdateAdapter {
    int ROWS = 15;
    int COLUMNS = 10;

    @BindView(R.id.lists)
    RecyclerView mTable;
    @BindView(R.id.restart)
    Button mRestart;
    @BindView(R.id.click)
    FrameLayout mClick;
    Adapter mAdapter;
    Presenter mPresenter;
    ProgressDialog loading;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_NOSENSOR);
        ButterKnife.bind(this);
//        loading = new ProgressDialog(this);
//        loading.setIndeterminate(true);
//        loading.setCancelable(false);
        mPresenter = new Presenter(ROWS, COLUMNS, this);

        RecyclerView.LayoutManager mLayoutManager = new GridLayoutManager(this, COLUMNS);
        mTable.setLayoutManager(mLayoutManager);
        mTable.setNestedScrollingEnabled(false);
        mPresenter.primaryState();
    }

    @OnClick(R.id.restart)
    public void onMRestartClicked() {
        mPresenter.primaryState();

    }

    @Override
    public void updateAdapter(KotlinField field) {
        //mTable.setAdapter(mAdapter);
        mTable.getAdapter().notifyDataSetChanged();
       // loading.dismiss();
    }

    @Override
    public void createAdapter(KotlinField field) {
        mAdapter = new Adapter(field, this, ROWS, COLUMNS);
        mTable.setAdapter(mAdapter);
        mTable.setHasFixedSize(true);
        mTable.getAdapter().notifyDataSetChanged();
    }


    @OnClick(R.id.click)
    public void onViewClicked() {
       // loading.show();
        mPresenter.nextStep();
    }


    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/lence/penguinsvskillerwhales/view/MainActivity.kt
package com.lence.penguinsvskillerwhales.view

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Button
import android.widget.FrameLayout
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.lence.penguinsvskillerwhales.R
import com.lence.penguinsvskillerwhales.life.Presenter

class MainActivity : AppCompatActivity(), UpdateAdapter {
    internal var ROWS = 15
    internal var COLUMNS = 10

    @BindView(R.id.lists)
    lateinit var mTable: RecyclerView
    @BindView(R.id.restart)
    lateinit var mRestart: Button
    @BindView(R.id.click)
    lateinit var mClick: FrameLayout
    internal var mAdapter: Adapter? = null
    internal var mPresenter: Presenter? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_NOSENSOR
        ButterKnife.bind(this)
        //        loading = new ProgressDialog(this);
        //        loading.setIndeterminate(true);
        //        loading.setCancelable(false);
        mPresenter = Presenter(ROWS, COLUMNS, this)

        val mLayoutManager = GridLayoutManager(this, COLUMNS)
        mTable.layoutManager = mLayoutManager
        mTable.isNestedScrollingEnabled = false
        mPresenter?.primaryState()
    }

    @OnClick(R.id.restart)
    fun onMRestartClicked() {
        mPresenter?.primaryState()

    }

    override fun updateAdapter(field: Field) {
        //mTable.setAdapter(mAdapter);
        mTable.adapter.notifyDataSetChanged()
        // loading.dismiss();
    }

    override fun createAdapter(field: Field) {
        mAdapter = Adapter(field, this, ROWS, COLUMNS)
        mTable.adapter = mAdapter
        mTable.setHasFixedSize(true)
        mTable.adapter.notifyDataSetChanged()
    }


    @OnClick(R.id.click)
    fun onViewClicked() {
        // loading.show();
        mPresenter?.nextStep()
    }


    override fun onStop() {
        super.onStop()
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
    }
}
